### PR TITLE
Make doctest SKIP option register doctests as SKIPPED rather than PASSED

### LINKED ISSUE #957: # doctest: +SKIP is registered as a test pass rather than a SKIP
`# doctest: +SKIP` is registered as a test pass rather than a SKIP

see also https://github.com/pytest-dev/pytest/issues/310

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #957

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
needs feedback from a actual user of doctests

**Comment by nicoddemus** [comment]
Good point!

Perhaps @graingert, @jaraco or @dimazest can chime in here (all users who have commented/posted issues about doctest)? My experience with doctest is really limited, so feedback here is welcome!

**Comment by The-Compiler** [comment]
Maybe @cpcloud @HolgerPeters @danilobellini @witsch as well - those are people who edited `_pytest/doctest.py` according to the git log.

File: testing/test_doctest.py [hunk 1]
```diff
@@ -446,3 +446,54 @@ def test_unicode_string(self, testdir):
...
+        makedoctest("""
+            >>> 1 + 1  # doctest: +SKIP
+            2
+            >>> 2 + 2  # doctest: +SKIP
+            200
+        """)
+        reprec = testdir.inline_run("--doctest-modules")
+        reprec.assertoutcome(skipped=1)
+
+
```
Comment (graingert) [inline testing/test_doctest.py]: ?

File: testing/test_doctest.py [hunk 2]
```diff
@@ -446,3 +446,54 @@ def test_unicode_string(self, testdir):
...
+    def test_one_skipped(self, testdir, makedoctest):
+        makedoctest("""
+            >>> 1 + 1  # doctest: +SKIP
+            2
+            >>> 2 + 2
+            4
+        """)
+        reprec = testdir.inline_run("--doctest-modules")
+        reprec.assertoutcome(passed=1)
```
Comment (graingert) [inline testing/test_doctest.py]: no way of getting this to be `passed=1, skipped=1`?

Comment (nicoddemus) [inline testing/test_doctest.py]: Unless I'm mistaken, currently `pytest` only considers the entire doc test block as passed/failed/skipped.

Changing `pytest` handling of doctests to split individual asserts into different test items would be out of scope for this PR, IMO.

File: testing/test_doctest.py (hunk 1 above)
Comment (nicoddemus) [inline testing/test_doctest.py]: Thanks!

**Comment by nicoddemus** [comment]
Rebased branch into `master` to get rid of the merge conflict.

**Comment by graingert** [comment]
This looks awesome! Thanks for the speedy turnaround!

**Pull request merged**